internal/fs/fs.go

```diff
@@ -1675,7 +1675,12 @@ func (fs *fileSystem) createLocalFile(
 
 	fullName := inode.NewFileName(parent.Name(), name)
 	child, ok := fs.localFileInodes[fullName]
-	if !ok {
+	// Create a new inode if a local file is unlinked and then recreated with the same name.
+	var fileInode *inode.FileInode
+	if child != nil {
+		fileInode = fs.fileInodeOrDie(child.ID())
+	}
+	if !ok || (fileInode != nil && fileInode.IsUnlinked()) {
```

Comment:
Can you simply do this: 
if !ok || child.(*inode.FileInode).IsUnlinked()

Reply:
What if child is nil? Shouldn't we keep `if !ok || (child!=nil && child.(*inode.FileInode).IsUnlinked())` ?